server/server.js

```diff
@@ -102,6 +103,23 @@ function validateBoardName(boardName) {
 function handleRequest(request, response) {
   var parsedUrl = new URL(request.url, 'http://wbo/');
   var parts = parsedUrl.pathname.split("/");
+
+  // Check for auth if required
+  if(config.AUTH_METHOD == "jwt") {
+    if(parsedUrl.searchParams.get("token")) {
+      var token = parsedUrl.searchParams.get("token");
+      jsonwebtoken.verify(token, config.AUTH_SECRET_KEY, function(error, decoded) {
+        if(error) { // Token not valid
+          response.writeHead(403, { 'Content-Type': 'text/plain' });
+          response.end("Error: Forbidden\n" + error);
```

Comment:
I don't think this code is correct. The verification is executed asynchronously, and does not prevent the rest of the code from executing. There is a race condition between the token verification and the happy path.

Reply:
You're totally correct, I've resolved this in the latest commit